src/wp-includes/rest-api/endpoints/class-wp-rest-menu-items-controller.php

```diff
@@ -101,6 +101,22 @@ protected function check_has_read_only_access( $request ) {
 		);
 	}
 
+	/**
+	 * Retrieves a collection of menu items.
+	 *
+	 * @since 6.x.x
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
```

Comment:
It's possible for a theme or plugin developer to set `suppress_filters => true` via the `rest_{$this->post_type}_query` filter, in which case this won't run.

There probably isn't much that can be done about this but it's worth noting anyway.

Reply:
We can move the actual function call into `WP_Query` which I'm still partial to instead of akwardly hooking in like this. https://github.com/WordPress/wordpress-develop/pull/2631#issuecomment-1139745739